src/config.coffee

```diff
@@ -1,41 +1,349 @@
+
+		@schema = {
+			apiEndpoint: { source: 'config.json' }
+			apiTimeout: { source: 'config.json', default: 15 * 60 * 1000 }
+			listenPort: { source: 'config.json', default: 48484 }
+			deltaEndpoint: { source: 'config.json', default: 'https://delta.resin.io' }
+			uuid: { source: 'config.json', mutable: true }
+			apiKey: { source: 'config.json', mutable: true, removeIfNull: true }
+			deviceApiKey: { source: 'config.json', mutable: true }
+			deviceType: { source: 'config.json', default: 'raspberry-pi' }
+			username: { source: 'config.json' }
+			userId: { source: 'config.json' }
+			deviceId: { source: 'config.json', mutable: true }
+			registered_at: { source: 'config.json', mutable: true }
+			applicationId: { source: 'config.json' }
+			appUpdatePollInterval: { source: 'config.json', mutable: true, default: 60000 }
+			pubnubSubscribeKey: { source: 'config.json', default: @constants.defaultPubnubSubscribeKey }
+			pubnubPublishKey: { source: 'config.json', default: @constants.defaultPubnubPublishKey }
+			mixpanelToken: { source: 'config.json', default: @constants.defaultMixpanelToken }
+			bootstrapRetryDelay: { source: 'config.json', default: 30000 }
+			supervisorOfflineMode: { source: 'config.json', default: false }
+			hostname: { source: 'config.json', mutable: true }
+
+			version: { source: 'func' }
+			currentApiKey: { source: 'func' }
+			offlineMode: { source: 'func' }
+			pubnub: { source: 'func' }
+			resinApiEndpoint: { source: 'func' }
+			provisioned: { source: 'func' }
+			osVersion: { source: 'func' }
+			osVariant: { source: 'func' }
+			provisioningOptions: { source: 'func' }
+			mixpanelHost: { source: 'func' }
+			extendedEnvOptions: { source: 'func' }
+			fetchOptions: { source: 'func' }
+
+			apiSecret: { source: 'db', mutable: true }
+			logsChannelSecret: { source: 'db', mutable: true }
+			name: { source: 'db', mutable: true }
+			initialConfigReported: { source: 'db', mutable: true, default: 'false' }
+			initialConfigSaved: { source: 'db', mutable: true, default: 'false' }
+			containersNormalised: { source: 'db', mutable: true, default: 'false' }
+			localMode: { source: 'db', mutable: true, default: 'false' }
+			loggingEnabled: { source: 'db', mutable: true, default: 'true' }
+			connectivityCheckEnabled: { source: 'db', mutable: true, default: 'true' }
+			delta: { source: 'db', mutable: true, default: 'false' }
+			deltaRequestTimeout: { source: 'db', mutable: true, default: '30000' }
+			deltaApplyTimeout: { source: 'db', mutable: true, default: '' }
+			deltaRetryCount: { source: 'db', mutable: true, default: '30' }
+			deltaRetryInterval: { source: 'db', mutable: true, default: '10000' }
+			lockOverride: { source: 'db', mutable: true, default: 'false' }
+			legacyAppsPresent: { source: 'db', mutable: true, default: 'false' }
+		}
+
+		@configJsonCache = {}
+
+		@_lock = new Lock()
+		@_writeLock = Promise.promisify(@_lock.async.writeLock)
+		@writeLockConfigJson = =>
+			@_writeLock('config.json')
```

Comment:
If you change this to just 
```coffeee
_writeLock('config.json')
```
then you avoid any risk of exposing it (I assume the `_` is to denote it's meant to be private) and can lose the fat arrow  as well